### DISCUSSION TITLE
Edit favorite routes

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#336] [Q&A] [answered]
If I want to edit a saved favorite route, I cannot simply overwrite the old one..
I have to save it as a new route.->Suboptimal...

**Comment by devemux86** [comment]
@Leeb64 thanks for the suggestion.

We probably need a function to update an existing favorite route, instead of creating a new one.

**Comment by devemux86** [comment]
How about the following workflow to edit favorite routes:

- When we show a favorite route on map, its title is marked as "active"

- When we long press the route line, the "Save favorite route" saves the changes updating the favorite route
  - The current "Add favorite route" is to add the route as new entry in favorites

- Which favorite route is "active" is cleared when we use the "Routing | Clear" menu

<img src="https://github.com/devemux86/cruiser/assets/3484020/5c9ea1d9-a23a-4b7d-99c0-cd0cbab65188" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/5fb1eddd-9d92-42a5-aef8-aa054c8f4618" width="50%"/>

**Reply by Leeb64** [reply]
Sounds good.

**Reply by DolomitesTom** [reply]
very 👍

**Reply by bios4** [reply]
> When we long press the route line, the "Save favorite route" saves the changes updating the favorite route
> The current "Add favorite route" is to add the route as new entry in favorites

What about naming it "Update saved favorite" in option 1, and "Save as new favorite" in option 2?

**Reply by devemux86** [reply]
Or we could name them:
- Add favorite route
- Update favorite route

**Comment by devemux86** [comment, marked as answer]
Edit favorite routes is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
